sm/plugins/tedge_apt_plugin/src/main.rs

```diff
@@ -73,27 +98,33 @@ fn run(operation: PluginOp) -> Result<ExitStatus, InternalError> {
             status
         }
 
-        PluginOp::Install { module, version } => {
-            if let Some(version) = version {
-                run_cmd(
-                    "apt-get",
-                    &format!("install --quiet --yes {}={}", module, version),
-                )?
-            } else {
-                run_cmd("apt-get", &format!("install --quiet --yes {}", module))?
+        PluginOp::Install { module, version } => install(module, version)?,
+
+        PluginOp::Remove { module, version } => remove(module, version)?,
+
+        PluginOp::UpdateList => {
+            let mut updates: Vec<SoftwareModuleUpdate> = Vec::new();
+            let mut status: ExitStatus = ExitStatus::from_raw(0);
+            let mut rdr = csv::ReaderBuilder::new()
+                .has_headers(false)
+                .delimiter(b' ')
+                .flexible(true)
```

Comment:
This highlights the main confusion we have about the expected input for this list of updates.

* In no case the proposal from @cstoidner was to use "*space* separated values2, but command line arguments where a non-empty sequence of spaces acts as a separator and quotes as grouper.
* Using a non-flexible csv format will be easier here. With a separator that is not the space character though.